dex/src/main/scala/com/wavesplatform/dex/market/ExchangeTransactionBroadcastActor.scala

```diff
@@ -25,20 +25,26 @@ class ExchangeTransactionBroadcastActor(settings: ExchangeTransactionBroadcastSe
 
   private val default: Receive = {
     case ExchangeTransactionCreated(tx) =>
-      check(tx) match {
+      try check(tx) match {
         case Right(_) => broadcast(List(tx))
         case Left(e)  => logError(tx, e)
+      } catch {
+        case e: Throwable => log.error(s"Can't check or broadcast the transaction ${tx.id()}", e)
       }
   }
 
   private def watching(toCheck: Vector[ExchangeTransaction], next: Vector[ExchangeTransaction]): Receive = {
     case ExchangeTransactionCreated(tx) =>
-      check(tx) match {
+      try check(tx) match {
```

Comment:
Is it possible to use `Try`?

Reply:
Sure, with more ~pain~ code :D 

```scala
Try(check(tx)) match {
  case Success(Right(_)) =>
    broadcast(List(tx))
    context.become(watching(toCheck, next :+ tx))
  
  case Success(Left(e)) => logError(tx, e)
  
  case Failure(e) =>
    log.warn(s"Can't check or broadcast the transaction ${tx.id()}, will try later", e)
    context.become(watching(toCheck, next :+ tx))
}

// or better

Try(check(tx))
  .map {
    case Right(_) => broadcast(List(tx)); true
    case Left(e) => logError(tx, e); false
  }
  .recover { case e => log.warn(s"Can't check or broadcast the transaction ${tx.id()}, will try later", e); true}
  .foreach { watchNext => if (watchNext) context.become(watching(toCheck, next :+ tx)) }
```

The second option is not so bad 🤔

Reply:
I decided to not to do this changes, because ExchangeTransactionBroadcastActor has been totally rewritten in master branch. With `try` keyword we will see less differences between current file and changes in `version-1.1.x`, so merging will be easier.